### ISSUE TITLE
[TASK] Fix CVE issues for v1.7.0 (RC3) 

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#9172]
## What's the task? Please describe

https://github.com/longhorn/longhorn/issues/8997#issuecomment-2267605744

**Comment by c3y1huang** [comment]
## Analyse v1.7.x-head

### Longhorn Components

#### longhornio/backing-image-manager:v1.7.x-head
```
longhornio/backing-image-manager:v1.7.x-head (suse linux enterprise server 15.6)
================================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)
```

Action: Won't fix.

#### longhornio/longhorn-engine:v1.7.x-head
```
longhornio/longhorn-engine:v1.7.x-head (suse linux enterprise server 15.6)
==========================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)
```

Action: Won't fix.

#### longhornio/longhorn-instance-manager:v1.7.x-head
```
longhornio/longhorn-instance-manager:v1.7.x-head (suse linux enterprise server 15.6)
====================================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)
```

Action: Won't fix.

#### longhornio/longhorn-manager:v1.7.x-head
```
longhornio/longhorn-manager:v1.7.x-head (suse linux enterprise server 15.6)
===========================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)


usr/local/sbin/longhorn-manager (gobinary)
==========================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌──────────────────────────────────────────────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────────┐
│                           Library                            │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                            │
├──────────────────────────────────────────────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ go.opentelemetry.io/contrib/instrumentation/google.golang.o- │ CVE-2023-47108 │ HIGH     │ fixed  │ v0.42.0           │ 0.46.0        │ opentelemetry-go-contrib: DoS vulnerability in otelgrpc due │
│ rg/grpc/otelgrpc                                             │                │          │        │                   │               │ to unbound cardinality metrics                              │
│                                                              │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2023-47108                  │
└──────────────────────────────────────────────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴─────────────────────────────────────────────────────────────┘
```

The `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is an indirect dependency, and the fixed version is not included in the current latest Kubernetes [v1.30.3](https://github.com/kubernetes/kubernetes/blob/v1.30.3/go.mod#L68) release. Its included in the incoming [v1.31.0](https://github.com/kubernetes/kubernetes/blob/v1.31.0-beta.0/go.mod#L206) release.

Action: Won't fix.

#### longhornio/longhorn-share-manager:v1.7.x-head
```
longhornio/longhorn-share-manager:v1.7.x-head (suse linux enterprise server 15.6)
=================================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)
```

Action: Won't fix.

#### longhornio/longhorn-ui:v1.7.x-head
```
longhornio/longhorn-ui:v1.7.x-head (suse linux enterprise server 15.6)
======================================================================
Total: 0 (HIGH: 0, CRITICAL: 0)
```

Action: Won't fix.

#### longhornio/support-bundle-kit:v0.0.40
```
longhornio/support-bundle-kit:v0.0.40 (suse linux enterprise server 15.6)
=========================================================================
Total: 5 (HIGH: 5, CRITICAL: 0)

┌────────────────────────────┬─────────────────────┬──────────┬────────┬────────────────────┬─────────────────────┬───────────────────────────────┐
│          Library           │    Vulnerability    │ Severity │ Status │ Installed Version  │    Fixed Version    │             Title             │
├────────────────────────────┼─────────────────────┼──────────┼────────┼────────────────────┼─────────────────────┼───────────────────────────────┤
│ libopenssl-3-fips-provider │ SUSE-SU-2024:2635-1 │ HIGH     │ fixed  │ 3.1.4-150600.5.7.1 │ 3.1.4-150600.5.10.1 │ Security update for openssl-3 │
├────────────────────────────┤                     │          │        │                    │                     │                               │
│ libopenssl3                │                     │          │        │                    │                     │                               │
├────────────────────────────┼─────────────────────┤          │        ├────────────────────┼─────────────────────┼───────────────────────────────┤
│ login_defs                 │ SUSE-SU-2024:2630-1 │          │        │ 4.8.1-150600.15.45 │ 4.8.1-150600.17.3.1 │ Security update for shadow    │
├────────────────────────────┼─────────────────────┤          │        ├────────────────────┼─────────────────────┼───────────────────────────────┤
│ openssl-3                  │ SUSE-SU-2024:2635-1 │          │        │ 3.1.4-150600.5.7.1 │ 3.1.4-150600.5.10.1 │ Security update for openssl-3 │
├────────────────────────────┼─────────────────────┤          │        ├────────────────────┼─────────────────────┼───────────────────────────────┤
│ shadow                     │ SUSE-SU-2024:2630-1 │          │        │ 4.8.1-150600.15.45 │ 4.8.1-150600.17.3.1 │ Security update for shadow    │
└────────────────────────────┴─────────────────────┴──────────┴────────┴────────────────────┴─────────────────────┴───────────────────────────────┘
```

Action: release v0.0.41

### External Components

#### longhornio/csi-attacher:v4.6.1
```
longhornio/csi-attacher:v4.6.1 (debian 12.5)
============================================
Total: 0 (HIGH: 0, CRITICAL: 0)


csi-attacher (gobinary)
=======================
Total: 1 (HIGH: 0, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.22.3            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

The current highest available minor version upstream: `registry.k8s.io/sig-storage/csi-attacher:v4.6.1`.

Action: Won't fix.

#### longhornio/csi-provisioner:v4.0.1
```
longhornio/csi-provisioner:v4.0.1 (debian 11.9)
===============================================
Total: 0 (HIGH: 0, CRITICAL: 0)


csi-provisioner (gobinary)
==========================
Total: 2 (HIGH: 1, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.21.5            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
│         ├────────────────┼──────────┤        │                   ├─────────────────┼────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45288 │ HIGH     │        │                   │ 1.21.9, 1.22.2  │ golang: net/http, x/net/http2: unlimited number of         │
│         │                │          │        │                   │                 │ CONTINUATION frames causes DoS                             │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2023-45288                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

The current highest available minor version upstream: `registry.k8s.io/sig-storage/csi-provisioner:v4.0.1`

Action: Won't fix.

#### longhornio/csi-resizer:v1.11.1
```
longhornio/csi-resizer:v1.11.1 (debian 12.5)
============================================
Total: 0 (HIGH: 0, CRITICAL: 0)

2024-08-05T00:58:05Z	WARN	Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.53/docs/scanner/vulnerability#severity-selection for details.

csi-resizer (gobinary)
======================
Total: 1 (HIGH: 0, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.22.3            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

The current highest available minor version: `registry.k8s.io/sig-storage/csi-resizer:v1.11.2`. However, this doesn't fix `CVE-2024-24790`.

Action: Won't fix.

#### longhornio/csi-snapshotter:v7.0.2
```
longhornio/csi-snapshotter:v7.0.2 (debian 11.9)
===============================================
Total: 0 (HIGH: 0, CRITICAL: 0)


csi-snapshotter (gobinary)
==========================
Total: 2 (HIGH: 1, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.21.5            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
│         ├────────────────┼──────────┤        │                   ├─────────────────┼────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45288 │ HIGH     │        │                   │ 1.21.9, 1.22.2  │ golang: net/http, x/net/http2: unlimited number of         │
│         │                │          │        │                   │                 │ CONTINUATION frames causes DoS                             │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2023-45288                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

The current highest available minor version upstream: `registry.k8s.io/sig-storage/csi-snapshotter:v7.0.2`

Action: Won't fix.

#### longhornio/csi-node-driver-registrar:v2.10.1
```
```

the current highest available minor version: `registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.11.1`.

Action: update to v2.11.1

#### longhornio/livenessprobe:v2.12.0
```
longhornio/csi-node-driver-registrar:v2.10.1 (debian 11.9)
==========================================================
Total: 0 (HIGH: 0, CRITICAL: 0)


csi-node-driver-registrar (gobinary)
====================================
Total: 2 (HIGH: 1, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.21.5            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
│         ├────────────────┼──────────┤        │                   ├─────────────────┼────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45288 │ HIGH     │        │                   │ 1.21.9, 1.22.2  │ golang: net/http, x/net/http2: unlimited number of         │
│         │                │          │        │                   │                 │ CONTINUATION frames causes DoS                             │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2023-45288                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

The current highest available minor version upstream: `registry.k8s.io/sig-storage/livenessprobe:v2.13.1`

Action: update to v2.13.1

#### longhornio/openshift-origin-oauth-proxy:4.15
```
longhornio/openshift-origin-oauth-proxy:4.15 (redhat 9.2)
=========================================================
Total: 8 (HIGH: 8, CRITICAL: 0)

┌────────────────────────┬────────────────┬──────────┬────────┬───────────────────┬──────────────────┬────────────────────────────────────────────────────────┐
│        Library         │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version   │                         Title                          │
├────────────────────────┼────────────────┼──────────┼────────┼───────────────────┼──────────────────┼────────────────────────────────────────────────────────┤
│ glibc                  │ CVE-2024-2961  │ HIGH     │ fixed  │ 2.34-60.el9_2.7   │ 2.34-60.el9_2.14 │ glibc: Out of bounds write in iconv may lead to remote │
│                        │                │          │        │                   │                  │ code...                                                │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-2961              │
│                        ├────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│                        │ CVE-2024-33599 │          │        │                   │                  │ glibc: stack-based buffer overflow in netgroup cache   │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-33599             │
├────────────────────────┼────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│ glibc-common           │ CVE-2024-2961  │          │        │                   │                  │ glibc: Out of bounds write in iconv may lead to remote │
│                        │                │          │        │                   │                  │ code...                                                │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-2961              │
│                        ├────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│                        │ CVE-2024-33599 │          │        │                   │                  │ glibc: stack-based buffer overflow in netgroup cache   │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-33599             │
├────────────────────────┼────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│ glibc-langpack-en      │ CVE-2024-2961  │          │        │                   │                  │ glibc: Out of bounds write in iconv may lead to remote │
│                        │                │          │        │                   │                  │ code...                                                │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-2961              │
│                        ├────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│                        │ CVE-2024-33599 │          │        │                   │                  │ glibc: stack-based buffer overflow in netgroup cache   │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-33599             │
├────────────────────────┼────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│ glibc-minimal-langpack │ CVE-2024-2961  │          │        │                   │                  │ glibc: Out of bounds write in iconv may lead to remote │
│                        │                │          │        │                   │                  │ code...                                                │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-2961              │
│                        ├────────────────┤          │        │                   │                  ├────────────────────────────────────────────────────────┤
│                        │ CVE-2024-33599 │          │        │                   │                  │ glibc: stack-based buffer overflow in netgroup cache   │
│                        │                │          │        │                   │                  │ https://avd.aquasec.com/nvd/cve-2024-33599             │
└────────────────────────┴────────────────┴──────────┴────────┴───────────────────┴──────────────────┴────────────────────────────────────────────────────────┘

usr/bin/oauth-proxy (gobinary)
==============================
Total: 6 (HIGH: 5, CRITICAL: 1)

┌──────────────────────────────────────────────────────────────┬─────────────────────┬──────────┬────────┬───────────────────┬──────────────────────────────────┬─────────────────────────────────────────────────────────────┐
│                           Library                            │    Vulnerability    │ Severity │ Status │ Installed Version │          Fixed Version           │                            Title                            │
├──────────────────────────────────────────────────────────────┼─────────────────────┼──────────┼────────┼───────────────────┼──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│ go.opentelemetry.io/contrib/instrumentation/google.golang.o- │ CVE-2023-47108      │ HIGH     │ fixed  │ v0.35.0           │ 0.46.0                           │ opentelemetry-go-contrib: DoS vulnerability in otelgrpc due │
│ rg/grpc/otelgrpc                                             │                     │          │        │                   │                                  │ to unbound cardinality metrics                              │
│                                                              │                     │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-47108                  │
├──────────────────────────────────────────────────────────────┼─────────────────────┤          │        ├───────────────────┼──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│ go.opentelemetry.io/contrib/instrumentation/net/http/otelht- │ CVE-2023-45142      │          │        │ v0.35.1           │ 0.44.0                           │ opentelemetry: DoS vulnerability in otelhttp                │
│ tp                                                           │                     │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45142                  │
├──────────────────────────────────────────────────────────────┼─────────────────────┤          │        ├───────────────────┼──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│ google.golang.org/grpc                                       │ GHSA-m425-mq94-257g │          │        │ v1.51.0           │ 1.56.3, 1.57.1, 1.58.3           │ gRPC-Go HTTP/2 Rapid Reset vulnerability                    │
│                                                              │                     │          │        │                   │                                  │ https://github.com/advisories/GHSA-m425-mq94-257g           │
├──────────────────────────────────────────────────────────────┼─────────────────────┼──────────┤        ├───────────────────┼──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│ stdlib                                                       │ CVE-2024-24790      │ CRITICAL │        │ 1.20.10           │ 1.21.11, 1.22.4                  │ golang: net/netip: Unexpected behavior from Is methods for  │
│                                                              │                     │          │        │                   │                                  │ IPv4-mapped IPv6 addresses                                  │
│                                                              │                     │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2024-24790                  │
│                                                              ├─────────────────────┼──────────┤        │                   ├──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│                                                              │ CVE-2023-45283      │ HIGH     │        │                   │ 1.20.11, 1.21.4, 1.20.12, 1.21.5 │ The filepath package does not recognize paths with a \??\   │
│                                                              │                     │          │        │                   │                                  │ prefix as...                                                │
│                                                              │                     │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45283                  │
│                                                              ├─────────────────────┤          │        │                   ├──────────────────────────────────┼─────────────────────────────────────────────────────────────┤
│                                                              │ CVE-2023-45288      │          │        │                   │ 1.21.9, 1.22.2                   │ golang: net/http, x/net/http2: unlimited number of          │
│                                                              │                     │          │        │                   │                                  │ CONTINUATION frames causes DoS                              │
│                                                              │                     │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45288                  │
└──────────────────────────────────────────────────────────────┴─────────────────────┴──────────┴────────┴───────────────────┴──────────────────────────────────┴─────────────────────────────────────────────────────────────┘
```

Action: not in scope, already updated to the suggested version: `quay.io/openshift/origin-oauth-proxy:4.15` (https://github.com/longhorn/longhorn/issues/8976#issuecomment-2230147085)

**Comment by c3y1huang** [comment]
## Action Summary
- longhornio/support-bundle-kit:
  - Release v0.0.41
- longhornio/csi-node-driver-registrar:
  - Update to v2.11.1
- longhornio/livenessprobe:
  - Update to v2.13.1

#### Won't Fix
- longhornio/backing-image-manager: 
  - No CVE vulnerability detected.
- longhornio/longhorn-engine:
  - No CVE vulnerability detected.
- longhornio/longhorn-instance-manager:
  - No CVE vulnerability detected.
- longhornio/longhorn-manager:
  - The `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is an indirect dependency, and the fixed version is not included in the current latest Kubernetes [v1.30.3](https://github.com/kubernetes/kubernetes/blob/v1.30.3/go.mod#L68) release. Its included in the incoming [v1.31.0](https://github.com/kubernetes/kubernetes/blob/v1.31.0-beta.0/go.mod#L206) release.
- longhornio/longhorn-share-manager:
  - No CVE vulnerability detected.
- longhornio/longhorn-ui:
  - No CVE vulnerability detected.
- longhornio/csi-attacher:
  - The current version is the same as the highest available minor version:`v4.6.1`.
- longhornio/csi-provisioner:
  - The current version is the same as the highest available minor version: `v4.0.1`.
- longhornio/csi-resizer:
  - The highest available minor version is `v1.11.2`, but it doesn't include the fix for `CVE-2024-24790`.
- longhornio/csi-snapshotter:
  - The current version is the same as the highest available minor version: `v7.0.2`
- longhornio/openshift-origin-oauth-proxy:
  - Not in scope, already updated to the suggested version: quay.io/openshift/origin-oauth-proxy:4.15 (https://github.com/longhorn/longhorn/issues/8976#issuecomment-2230147085)

**Comment by c3y1huang** [comment]
All PRs have been merged. This doesn't require effort from @longhorn/qa , any issues introduced should be detected in the daily regression run. Closing.
